src/games/strategy/engine/message/UnifiedMessengerHub.java

```diff
@@ -144,7 +144,9 @@ private void invoke(final HubInvoke hubInvoke, final Collection<INode> remote, f
   public void waitForNodesToImplement(final String endPointName) {
     final long endTime = NODE_IMPLEMENTATION_TIMEOUT + System.currentTimeMillis();
     while (System.currentTimeMillis() < endTime && !hasImplementors(endPointName)) {
-      ThreadUtil.sleep(50);
+      if(!ThreadUtil.sleep(50)) {
+        break;
```

Comment:
Better to have a return instead of break here? Functionally equivalent I believe.